openverse_catalog/dags/providers/provider_api_scripts/wordpress.py

```diff
@@ -10,287 +10,184 @@
-
-        return image_url, height, width, filetype, filesize
-    return None, None, None, None, None
-
-
-def _get_filesize(image_url):
-    resp = delayed_requester.get(image_url)
-    if resp:
-        filesize = int(resp.headers.get("Content-Length", 0))
-        return filesize if filesize != 0 else None
-
-
-def _get_author_data(image):
-    try:
-        raw_author = image.get("_embedded", {}).get("author", [])[0]
-    except IndexError:
-        return None, None
-    author = raw_author.get("name")
-    if author is None or author == "":
-        author = raw_author.get("slug")
-    author_url = raw_author.get("url")
-    if author_url == "":
-        author_url = raw_author.get("link")
-    return author, author_url
-
-
-def _get_title(image):
-    if title := image.get("content", {}).get("rendered"):
-        try:
-            title = html.fromstring(title).text_content()
-        except UnicodeDecodeError as e:
-            logger.warning(f"Can't save the image's title ('{title}') due to {e}")
-            return None
-    return title
-
-
-def _get_metadata(media_data, media_details):
-    raw_metadata = media_details.get("image_meta", {})
-    metadata, tags = {}, []
-    extras = [
+class WordPressDataIngester(ProviderDataIngester):
+    host = "wordpress.org"
+    endpoint = f"https://{host}/photos/wp-json/wp/v2/photos"
+    providers = {constants.IMAGE: prov.WORDPRESS_DEFAULT_PROVIDER}
+    license_url = "https://creativecommons.org/publicdomain/zero/1.0/"
+    metadata_extras = [
         "aperture",
         "camera",
         "created_timestamp",
         "focal_length",
         "iso",
         "shutter_speed",
     ]
-    for key in extras:
-        value = raw_metadata.get(key)
-        if value not in [None, ""]:
-            metadata[key] = value
-
-    raw_related_resources = media_data.get("_embedded", {}).get("wp:term", [])
-    resource_mapping = {
+    metadata_resource_mapping = {
         "photo_category": "categories",
         "photo_color": "colors",
         "photo_orientation": "orientation",
         "photo_tag": "tags",
     }
-    for resource_arr in raw_related_resources:
-        for resource in resource_arr:
-            if (txy := resource.get("taxonomy")) in resource_mapping.keys():
-                resource_key = resource_mapping[txy]
-                resource_val = resource.get("name")
-                if txy == "photo_tag":
-                    tags.append(resource_val)
-                elif txy == "photo_orientation":
-                    metadata["orientation"] = resource_val
-                else:
-                    metadata.setdefault(resource_key, [])
-                    metadata[resource_key].append(resource_val)
-    return metadata, tags
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.license_info = get_license_info(license_url=self.license_url)
+
+    def get_media_type(self, record: dict) -> str:
+        return constants.IMAGE
```

Comment:
If you rebase now that #584 is merged you could delete this method :grin:

Reply:
Ooo great call, will do!